cedar-policy-validator/src/types.rs

```diff
@@ -904,9 +1022,40 @@ impl EntityRecordKind {
         }
     }
 
+    /// Return `true` if this entity or record may have additional undeclared
+    /// attributes.
+    pub(crate) fn has_open_attributes_record(&self, schema: &ValidatorSchema) -> bool {
+        match self {
```

Comment:
Redundant with code that seems to be doing the same thing above, i.e., `may_have_attr` ? Repeating comment from there: What about `Never` ?